yarn-project/noir-libs/noir-aztec/src/types/vec.nr

```diff
@@ -37,18 +37,46 @@ impl<T, MaxLen> BoundedVec<T, MaxLen> {
         self.len -= 1;
         elem
     }
+
+    fn any<Env>(self, predicate: fn[Env](T) -> bool) -> bool {
+        let mut ret = false;
+        for i in 0..MaxLen {
+            if (i < self.len as u64) {
+                ret |= predicate(self.storage[i]);
+            }
+        }
+        ret
+    }
+}
+
+#[test]
+fn test_vec_push_pop() {
+    let mut vec: BoundedVec<Field, 3> = BoundedVec::new(0);
+    assert(vec.len == 0);
+    vec.push(2);
+    assert(vec.len == 1);
+    vec.push(4);
+    assert(vec.len == 2);
+    vec.push(6);
+    assert(vec.len == 3);
+    let x = vec.pop();
+    assert(vec.len == 2);
+    assert(x == 6);
+}
+
+#[test(should_fail)]
+fn test_vec_push_overflow() {
+    let mut vec: BoundedVec<Field, 1> = BoundedVec::new(0);
+    vec.push(1);
+    vec.push(2);
```

Comment:
I have a feeling Noir tests now support failure assertions

Reply:
Unfortunately not yet :(